platforms/nuttx/src/px4/common/px4_init.cpp

```diff
@@ -42,7 +42,61 @@
 
 #include <fcntl.h>
 
-int px4_platform_console_init(void)
+#if defined(GPIO_OTGFS_VBUS) && defined(CONFIG_SYSTEM_CDCACM)
+__BEGIN_DECLS
+#include <nuttx/wqueue.h>
+#include <builtin/builtin.h>
+
+static struct work_s usb_serial_work;
+
+extern int sercon_main(int c, char **argv);
+extern int serdis_main(int c, char **argv);
+
+static void mavlink_usb_start(void *arg)
+{
+	if (sercon_main(0, NULL) == EXIT_SUCCESS) {
+		sched_lock();
```

Comment:
Why lock the whole system? Is this to avoid bounce?  It would be batter to Better gate and debounce the ISR

Reply:
It's not a debounce, currently just replicating what `nsh_builtin()` is doing (lock + exec_builtin), the lock may not be necessary.

Reply:
some of my PixRacers behave strange only with this PR: sensors status shows 0Hz
or only half of the imu are shown 
https://review.px4.io/plot_app?log=60c79a94-4e4f-4aa7-a4dd-9e0d70360e91

sensors status INFO  [sensors] selected gyro: 0 (0) INFO  [data_validator] validator: best: -1, prev best: -1, failsafe: NO (0 events)  INFO  [sensors] selected accel: 0 (0) INFO  [data_validator] validator: best: -1, prev best: -1, failsafe: NO (0 events)

 INFO  [vehicle_acceleration] selected sensor: 0, rate: nan Hz, estimated bias: [0.0000 0.0000 0.0000] INFO  [sensor_calibration] ACC 0 EN: 1, offset: [0.0000 0.0000 0.0000] scale: [1.0000 1.0000 1.0000]  INFO  [vehicle_angular_velocity] selected sensor: 0, rate: nan Hz  INFO  [vehicle_angular_velocity] estimated bias: [0.0000 0.0000 0.0000] INFO  [sensor_calibration] GYRO 0 EN: 1, offset: [0.0000 0.0000 0.0000]

Reply:
@taileron I think that might have been an unrelated bug that was briefly in master. https://github.com/PX4/PX4-Autopilot/pull/17776

I'll get the PR rebased.